frontend/src/components/ProjectForm/ProjectForm.tsx

```diff
@@ -83,9 +78,22 @@ export default function ProjectForm() {
   const [validRunner, setValidRunner] = useState(true);
   const [validSubmission, setValidSubmission] = useState(true);
 
-  useEffect(() => {
-    fetchCourses();
-  }, [regexError]);
+  const courses = useLoaderData() as Course[]
+
+  const location = useLocation();
+  const urlParams = new URLSearchParams(location.search);
+  const initialCourseId = urlParams.get('course_id') || '';
+  let initialCourseName = ''
+  for( const c of courses){
```

Comment:
Wrap this in an effect, or this will be run on every key/mouse pressed on this page